### ISSUE TITLE
Misaligned two-qubit gate diagrams for deep circuits

### ISSUE THREAD

**Issue by andbe91** [quantumlib/Cirq#5719]
**Description of the issue**
Lines connecting two qubits get gradually misaligned for deep circuits in Google colab (with a supposedly monospaced font).
<img width="1761" alt="image" src="https://user-images.githubusercontent.com/23558186/178350380-2b5d57d2-ee28-4e03-be15-53a68b6ae901.png">

**How to reproduce the issue**

```python
import cirq
q0, q1 = cirq.LineQubit.range(2)
circuit = cirq.Circuit([cirq.CZ(q0, q1)] * 200)
print(circuit)
```

**Cirq version**
0.15.0

**Comment by andbe91** [comment]
@maffoo yeah I agree. Seems to be a macOs and/or Colab issue since I only see this is on my MacBook and not on my linux desktop. So can probably close this issue

**Comment by Batmaev** [comment]
Actual strings are aligned.

The problem occurs because the output is inside the `<pre>` HTML element. By default, it has the property `font-family: monospace`. Which monospace font will be used depends on the browser and OS. 

On macOS, Chromium-based browsers seem to select the old Courier font, which lacks box-drawing characters  `│` and `─`. These characters get displayed in Menlo, which is also monospaced but has a different character width.

The solution is to change the font to one that supports box drawing. Menlo, which is the default monospace font in Firefox for macOS, might be a good option.

**Comment by Batmaev** [comment]
The problem occurs in two scenarios:

1.
```python
circuit = cirq.Circuit([cirq.CZ(q0, q1)] * 200)
print(circuit) # calls circuit.__str__

# <misaligned output>
```
In the first scenario, we can't do anything since `circuit.__str__()` must return plain text. We can only contact Colab or Chrome developers and ask them to change the font.

2.
```python
circuit = cirq.Circuit([cirq.CZ(q0, q1)] * 200)
circuit        # calls circuit._repr_html_

# <misaligned output>
```

The second scenario produces misaligned output not only in Colab on Chrome but also in Safari and VS Code on macOS.

Fortunately, here we have control over font.

For example, instead of this:

https://github.com/quantumlib/Cirq/blob/65ebb4a4d4a7b8b15e766b486e19582c634c1641/cirq-core/cirq/circuits/circuit.py#L290-L296

we can write this:
```python
def _repr_html_(self) -> str:
    """Print ASCII diagram in Jupyter notebook without wrapping lines."""
    return (
        '<pre style="overflow: auto; white-space: pre; font-family: Menlo, monospace;">'
        + html.escape(self.to_text_diagram())
        + '</pre>'
    )
```
Almost all Mac users have the Menlo font preinstalled and it will be used for them.
Most non-Mac users don't have this font. For them, the browser-default monospace font will be used, as before.

This will fix the problem in the second scenario.

**Comment by dstrain115** [comment]
This seems like a good first issue for anyone with a mac device that can test.

**Comment by mhucka** [comment]
I've tried to reproduce this, but have not been able to. Here are the cases:

1. Google Colab (version of 2025-04-09), running on MacOS 15.4 (24E248), Cirq version 1.5.0: using the default Colab font ("monospace") at 14 points, in Google Chrome version 135.0.7049.95, I see this:

![Image](https://github.com/user-attachments/assets/2271f6f4-64b5-4b2d-83b1-c1f8d0ff5d24)

2. Python 3.10.14 REPL in iTerm build 3.5.11, on MacOS 15.4 (24E248), using the font "UbuntuMono Nerd Font Mono" at 18 pts, I see this:

![Image](https://github.com/user-attachments/assets/98b7f559-e1c8-45ac-9b38-6ee75e396d48)

**Comment by mhucka** [comment]
As I have not been able to reproduce this, nor find someone in the team who could, I'm going to close this as something that has resolved itself in Colab since in the years since the issue was first file. If this is in error, someone please reopen this issue.
